### PR TITLE
update and migrate `descheduler` jobs

### PR DESCRIPTION
**Pull request by rjsadow**
This PR aims to provide four updates to the descheduler jobs.

1. remove unsuported kubernetes versions and their jobs
2. bump all kubernetes versions to their latest releases
3. add jobs for the 1.28 release
4. migrate all remaining jobs to the eks cluster

### CODE REVIEW

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml [hunk 1]
```diff
@@ -137,7 +137,7 @@ presubmits:
         - runner.sh
         env:
         - name: KUBERNETES_VERSION
-          value: "v1.26.3"
+          value: "v1.27.4"
```
Comment (ingvagabund) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml:140]: Not yet at https://hub.docker.com/r/kindest/node/tags?page=1&name=1.27

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml [hunk 2]
```diff
@@ -175,7 +175,7 @@ presubmits:
         - runner.sh
         env:
         - name: KUBERNETES_VERSION
-          value: "v1.25.8"
+          value: "v1.26.7"
```
Comment (ingvagabund) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml:178]: Not yet in https://hub.docker.com/r/kindest/node/tags?page=1&name=1.26

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml [hunk 3]
```diff
@@ -99,7 +99,7 @@ presubmits:
         - runner.sh
         env:
         - name: KUBERNETES_VERSION
-          value: "v1.27.0"
+          value: "v1.28.0"
```
Comment (ingvagabund) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml:102]: Available at https://hub.docker.com/layers/kindest/node/v1.28.0/images/sha256-9f3ff58f19dcf1a0611d11e8ac989fdb30a28f40f236f59f0bea31fb956ccf5c?context=explore

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.25.yaml [hunk 4]
```diff
@@ -74,7 +99,7 @@ presubmits:
         - runner.sh
         env:
         - name: KUBERNETES_VERSION
-          value: "v1.25.0"
+          value: "v1.25.12"
```
Comment (ingvagabund) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.25.yaml:102]: Not yet available at https://hub.docker.com/r/kindest/node/tags?page=1&name=1.25

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.25.yaml [hunk 5]
```diff
@@ -104,7 +137,7 @@ presubmits:
         - runner.sh
         env:
         - name: KUBERNETES_VERSION
-          value: "v1.24.3"
+          value: "v1.24.16"
```
Comment (ingvagabund) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.25.yaml:140]: Not yet available at https://hub.docker.com/r/kindest/node/tags?page=1&name=1.24

File: config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.25.yaml [hunk 6]
```diff
@@ -134,7 +175,7 @@ presubmits:
         - runner.sh
         env:
         - name: KUBERNETES_VERSION
-          value: "v1.23.4"
+          value: "v1.23.14"
```
Comment (ingvagabund) [inline config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-release-1.25.yaml:178]: 1.23.17 available at https://hub.docker.com/layers/kindest/node/v1.23.17/images/sha256-0597166bd392d108b0a80e8e12d817a2d597d9d2f1d10f48f95583f5e6167098?context=explore

**Comment by ingvagabund** [comment]
/hold
either until the missing versions are lowered or available
@rjsadow thanks for the heads-up

**Comment by rjsadow** [comment]
Thanks @ingvagabund! I've updated all the tags to match the latest images available.

**Pull request merged**